### PR TITLE
Decompose measurements to counts (with OQC integration)

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**
Devices supporting only counts do not currently decompose measurements.

**Description of the Change:**
We automatically add the transform to decompose a measurement to counts and post processing, when only Counts is defined on the TOML file.

**Benefits:**
Hardware devices support more measurements.

**Possible Drawbacks:**
Does not work with gradients/jacobian.

### CODE REVIEW

**Comment by rmoyard** [comment]
[sc-61993]

File: frontend/catalyst/preprocess.py [hunk 1]
```diff
@@ -230,6 +230,7 @@ def measurements_from_counts(tape):
 
     new_measurements = [qml.counts(wires=list(measured_wires))]
     new_tape = type(tape)(news_operations, new_measurements, shots=tape.shots)
+    print(new_tape.circuit)
```
Comment (rmoyard) [inline frontend/catalyst/preprocess.py:233]:
```suggestion
```

File: frontend/catalyst/qjit_device.py [hunk 2]
```diff
@@ -298,6 +307,10 @@ def preprocess(
         ops_acceptance = partial(catalyst_acceptance, operations=self.operations)
         program.add_transform(decompose, ctx=ctx, stopping_condition=ops_acceptance)
 
+        counts = {"Counts"}
+        if self.measurement_processes == counts:
```
Comment (dime10) [inline frontend/catalyst/qjit_device.py:311]:
```suggestion
        if self.measurement_processes == {"Counts"}:
```

File: frontend/catalyst/qjit_device.py [hunk 3]
```diff
@@ -298,6 +307,10 @@ def preprocess(
...
+            program.add_transform(measurements_from_counts)
```
Comment (erick-xanadu) [inline frontend/catalyst/qjit_device.py:312]: Should this be:

```suggestion
        counts = {"Counts"}
        if counts in self.measurement_processes:
            program.add_transform(measurements_from_counts)
```

File: frontend/test/pytest/test_preprocess.py [hunk 4]
```diff
@@ -171,6 +225,31 @@ def circuit(theta: float):
         assert "var" not in mlir
         assert "counts" in mlir
 
+    @pytest.mark.skipif(
+        not pathlib.Path(
+            get_lib_path("runtime", "RUNTIME_LIB_DIR") + "/libdummy_device.so"
+        ).is_file(),
+        reason="lib_dummydevice.so was not found.",
+    )
```
Comment (dime10) [inline frontend/test/pytest/test_preprocess.py:233]: Would be great to ensure this always runs, not only when the optional dummy component is available.

**Review by dime10 — APPROVED** [review]
Thanks looks great!

File: frontend/catalyst/qjit_device.py (hunk 3 above)
Comment (erick-xanadu) [reply frontend/catalyst/qjit_device.py:312]: I see where I was wrong.

File: frontend/test/pytest/test_preprocess.py (hunk 4 above)
Comment (rmoyard) [reply frontend/test/pytest/test_preprocess.py:233]: https://github.com/PennyLaneAI/catalyst/pull/702

Comment (rmoyard) [reply frontend/test/pytest/test_preprocess.py]: Merged

**Comment by erick-xanadu** [comment]
@rmoyard it looks like you only need rc to be merged back to main. Would this be a correct assessment on my part?

**Pull request merged**